api/v1alpha1/externaldns_types.go

```diff
@@ -427,14 +427,11 @@ type ExternalDNSSourceUnion struct {
 	// +required
 	Type ExternalDNSSourceType `json:"type"`
 
-	// AnnotationFilter describes an annotation filter
-	// used to filter which source instance resources
-	// ExternalDNS publishes records for.
-	// The annotation filter uses label selector semantics
-	// against source resource annotations.
+	// LabelFilter describes an annotation filter used to filter which source instance resources ExternalDNS publishes
+	// records for. The label filter uses label selector semantics against source resource annotations.
 	//
 	// +optional
-	AnnotationFilter map[string]string `json:"annotationFilter,omitempty"`
+	LabelFilter *string `json:"labelFilter,omitempty"`
```

Comment:
I think I'll stop here the review as the rest is pretty much dependent on this one.

I would like to challenge the type change. I think `map[string]string` is more appropriate as type. It brings UX similar to `metadata.labels` or any label selector.

@Miciah : WDYT?

Reply:
@alebedev87 @Miciah It is in fact implemented as `metav1.LabelSelector`. Any arbitrary selector can be passed to the controller.

Reply:
@Miciah just a observation. I see that both the `string` and `*metav1.LabelSelector` types are used in places. [Example 1](https://github.com/kubernetes/api/blob/872e73e89113b552c7677bb8e112b9ef7a61d826/apps/v1/types.go#L344) and [Example 2](https://github.com/kubernetes/apimachinery/blob/32df71429c452d9ef5e0290ee53932b39a9e4e36/pkg/apis/meta/v1/types.go#L342). As a user I would prefer to write the selector as a string because specifying it as a structed object is cumbersome. In fact the [docs](https://github.com/kubernetes/apimachinery/blob/1d209c836ad2fed6332e0aa08aefc669a649ba8e/pkg/apis/meta/v1/types.go#L1165) on `metav1.LabelSector` says _LabelSelector is an internal representation for the latter style._